src/Components/BookMark.js

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import { toast } from "react-toastify";
+import StyledMsg from "./styles/ToastMsg.styled";
+
+const BookMarkStyle = styled.svg`
+  position: absolute;
+  margin-left: ${(props) => (props.$isModal === true ? "1.5rem" : "14.25rem")};
+  margin-top: ${(props) => (props.$isModal === true ? "26.81rem" : "10.87rem")};
+  z-index: ${(props) => (props.$isModal === true ? 15 : 0)};
+  width: 1.5rem;
+  height: 1.5rem;
+  flex-shrink: 0;
+  filter: drop-shadow(2px 2px 4px rgba(0, 0, 0, 0.1));
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  .star {
+    margin-right: 0.5rem;
+  }
+`;
+const BookMark = ({ item, bookMarks, setBookMarks, $isModal }) => {
+  const [checked, setChecked] = useState(false);
+ 
+  const notify = (isAdd) => {
+    toast(<StyledMsg isAdd={isAdd}/>, {
+      position: "bottom-right",
+      autoClose: 5000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+      theme: "light",
+    });
+  };
+  const addBookMark = (item) => {
+    let arr = [];
+    if (JSON.parse(localStorage.getItem("bookmarks"))) {
+      arr = [...JSON.parse(localStorage.getItem("bookmarks"))];
+    }
```

Comment:
```suggestion
const bookmarked = JSON.parse(localStorage.getItem("bookmarks"))
    if (bookmarked) {
      arr = [...bookmarked];
    }
```

JSON.parse를 굳이 2번 할 필요 없어 코드 수정 제안합니다.